crates/matrix-sdk/src/widget/settings/element_call.rs

```diff
@@ -213,22 +257,53 @@ mod tests {
 
     const WIDGET_ID: &str = "1/@#w23";
 
-    fn get_widget_settings(encryption: Option<EncryptionSystem>) -> WidgetSettings {
-        WidgetSettings::new_virtual_element_call_widget(VirtualElementCallWidgetOptions {
+    fn get_widget_settings(
+        encryption: Option<EncryptionSystem>,
+        posthog: bool,
+        rageshake: bool,
+        sentry: bool,
+    ) -> WidgetSettings {
+        let props = VirtualElementCallWidgetOptions {
             element_call_url: "https://call.element.io".to_owned(),
             widget_id: WIDGET_ID.to_owned(),
-            parent_url: None,
             hide_header: Some(true),
             preload: Some(true),
-            font_scale: None,
             app_prompt: Some(true),
-            skip_lobby: Some(false),
             confine_to_room: Some(true),
-            font: None,
-            analytics_id: None,
             encryption: encryption.unwrap_or(EncryptionSystem::PerParticipantKeys),
-        })
-        .expect("could not parse virtual element call widget")
+            ..VirtualElementCallWidgetOptions::default()
+        };
+
+        let props = if posthog {
+            VirtualElementCallWidgetOptions {
+                posthog_user_id: Some("POSTHOG_USER_ID".to_owned()),
+                posthog_api_host: Some("posthog.element.io".to_owned()),
+                posthog_api_key: Some("POSTHOG_KEY".to_owned()),
+                ..props
+            }
+        } else {
+            props
+        };
```

Comment:
Why not make the original props mutable and then just assign them? 🤔  Seems simpler to read to me ..

```suggestion
        if posthog {
            props.posthog_user_id = Some("POSTHOG_USER_ID".to_owned());
            props.posthog_api_host = Some("posthog.element.io".to_owned());
            props.posthog_api_key = Some("POSTHOG_KEY".to_owned());
        }
```

Reply:
done